tests/CMakeLists.txt

```
@@ -2,6 +2,19 @@ add_compile_options("-Wno-undef")
 add_compile_options("-Wno-switch-default")
 add_compile_options("-Wno-switch-enum")
 
+# Architecture-specific preprocessor macro for *.cpp/*.c
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+  add_definitions(-DARCH_AARCH64)
```

I think it's a bit nicer to use the target_compile_definitions
```suggestion
  target_compile_definitions(bpftrace_test PRIVATE ARCH_AARCH64)
```

---

Actually 'target_compile_definitions' is used in the original patch, but failed with several automatic build test triggered by GitHub, warning that "target_compile_definitions is not defined". Maybe it's related to the version of cmake, since target_compile_definitions is added since cmake3?

---

Do you have a link to the failed builds? It should work. We use it in a quite a few places and we require cmake 3.something anyway:

```
tests/CMakeLists.txt
69:  target_compile_definitions(bpftrace_test PRIVATE HAVE_LIBBPF_MAP_BATCH)
73:  target_compile_definitions(bpftrace_test PRIVATE HAVE_NAME_TO_HANDLE_AT=1)
76:  target_compile_definitions(bpftrace_test PRIVATE HAVE_BCC_PROG_LOAD)
79:  target_compile_definitions(bpftrace_test PRIVATE HAVE_BCC_CREATE_MAP)
```

Maybe you placed it before the bpftrace_test target definition? Needs to go on line 69 or later

> I'm not familiar with cmake and really look forward you guys' professional opinion. Thanks.

We all struggle with cmake ;)